src/workerd/util/sqlite.c++

```diff
@@ -720,7 +722,33 @@ SqliteDatabase::StatementAndEffect SqliteDatabase::prepareSql(const Regulator& r
 
           // This isn't the last statement in the code. Execute it immediately.
           SQLITE_CALL_SCOPE {
+            auto start = sqliteObserver.now();
+            auto dbWalSizeBefore = sqliteObserver.getDbWalSize();
+
             int err = sqlite3_step(result);
+
+            kj::Duration queryLatency = sqliteObserver.now() - start;
+            auto dbWalBytesWritten = sqliteObserver.getDbWalSize() - dbWalSizeBefore;
+            auto rowsRead = sqlite3_stmt_status(result, LIBSQL_STMTSTATUS_ROWS_READ, 0);
+            auto rowsWritten = sqlite3_stmt_status(result, LIBSQL_STMTSTATUS_ROWS_WRITTEN, 0);
+
+            kj::Maybe<kj::String> queryStatement;
+            kj::Maybe<kj::String> queryErrorDescription;
+            try {
+              kj::StringPtr statement = sqlite3_sql(result);
+              queryStatement = kj::heapString(
+                  statement.slice(0, kj::min(statement.size(), RA_MAX_METRICS_QUERY_SIZE)));
+            } catch (kj::Exception& e) {
+              kj::StringPtr errorDescription = e.getDescription();
+              queryErrorDescription = kj::heapString(errorDescription.slice(
```

Comment:
Oh, good catch!  I could have sworn `kj::StringPtr::slice()` had a built-in `min()` behavior, but looking through the implementation, I see it just has debug-only bounds checks.